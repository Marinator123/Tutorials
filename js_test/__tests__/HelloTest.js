import React from 'react';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Hello from '../src/Hello.jsx'
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('Testet die Hello Komponente mit Jest Snapshot', () => {
    const component = renderer.create(
        <Hello labelOn="On" labelOff="Off" />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onChange();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Testet die Hello Checkbox mit Enzyme', () => {
    const checkbox = shallow(<Hello labelOn="On" labelOff="Off" />);
    
    expect(checkbox.text()).toEqual('HelloOff');
    checkbox.find('input').simulate('change');
    expect(checkbox.text()).toEqual('HelloOn');
});

