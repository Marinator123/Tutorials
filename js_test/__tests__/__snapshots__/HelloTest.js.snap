// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testet die Hello Komponente mit Jest Snapshot 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;

exports[`Testet die Hello Komponente mit Jest Snapshot 2`] = `
<label>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  On
</label>
`;
